docs/src/main/asciidoc/spring-cloud-commons.adoc

```diff
@@ -167,6 +167,8 @@ This feature addresses the problem of stateful beans that only get their configu
 For instance, if a `DataSource` has open connections when the database URL is changed via the `Environment`, you probably want the holders of those connections to be able to complete what they are doing.
 Then, the next time something borrows a connection from the pool, it gets one with the new URL.
 
+Sometimes, it might even be mandatory to apply the `@RefreshScope` annotation on some beans which can be only initialized once. For example, starting from Spring Boot 2, the default DataSource has become `com.zaxxer.hikari.HikariDataSource`, which by default is `refreshable` (defined in `RefreshScopeBeanDefinitionEnhancer`). It's because HikariDataSource can no longer be modified once the connection pool has been started. In order to refresh this kind of `immutable` beans, you will have to either annotate the beans with `@RefreshScope` or specify the list of classname(s) under the property key `spring.cloud.refresh.extra-refreshable`.
```

Comment:
I dont think we need this sentence
"It's because HikariDataSource can no longer be modified once the connection pool has been started."

Reply:
Well, I do :)

Without this explanation, users won't be able to understand why this hack is done in RefreshScopeBeanDefinitionEnhancer. Otherwise, users will have to do the same as me, dive into your code, and try to understand why RefreshScope is necessary.

However, I could try to shorten this sentence a bit, sth like:
For example, starting from Spring Boot 2, the default DataSource has become `com.zaxxer.hikari.HikariDataSource`, which by default is `refreshable` (defined in `RefreshScopeBeanDefinitionEnhancer`) because of immutability.

Reply:
I think it is good enough to just state that if a bean is "immutable" than you will need to annotate the beans or use the property.  Going into details about that one specific class isnt helping the user solve their problem.

Reply:
I agree with Ryan. I think my confusion arose from thinking the explanation was pointing to something I really need to know when it's really pointing to an implementation detail. It may be useful to know, but it needs to be a background detail.